enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyAuthenticator.java

```diff
@@ -55,17 +50,34 @@ public APIKeyAuthenticator() {
     @Override
     public boolean canAuthenticate(RequestContext requestContext) {
 
-        if (!isAPIKeyEnabled) {
-            return false;
-        }
         String apiKeyValue = getAPIKeyFromRequest(requestContext);
         return apiKeyValue != null && apiKeyValue.startsWith(APIKeyConstants.API_KEY_PREFIX);
     }
 
     @Override
     public AuthenticationContext authenticate(RequestContext requestContext) throws APISecurityException {
 
-        return super.authenticate(requestContext);
+        AuthenticationContext authCtx = super.authenticate(requestContext);
+        // Drop the API key data from the API key header.
+        dropAPIKeyDataFromAPIKeyHeader(requestContext);
+        return authCtx;
+    }
+
+    private void dropAPIKeyDataFromAPIKeyHeader(RequestContext requestContext) throws APISecurityException {
+
+        String apiKeyHeaderValue = getAPIKeyFromRequest(requestContext).trim();
+        String checksum = apiKeyHeaderValue.substring(apiKeyHeaderValue.length() - 6);
```

Comment:
Should we check the length and do processing as indexOutOfBound exception could be thrown.

Reply:
I guess we can process this in can authenticate method straightaway and then no need to check that repeatedly everywhere.